lib/qless/middleware/retry_exceptions.rb

```diff
@@ -48,19 +48,18 @@ def on_retry_callback
         @on_retry_callback ||= DEFAULT_ON_RETRY_CALLBACK
       end
 
-      def exponential(base, options = {})
+      # If`factor` is omitted it is set to `delay_s` to reproduce legacy
+      # behavior.
+      def exponential(delay_s, options={})
+        factor = options.fetch(:factor, delay_s)
         fuzz_factor = options.fetch(:fuzz_factor, 0)
 
-        lambda do |num, _error|
-          unfuzzed = base**num
+        lambda do |*unused_args|
+          unfuzzed = delay_s
+          delay_s *= factor
```

Comment:
BTW, given that this is an easy mistake to make, it would be to add a spec for this case.  I think you can just use `perform_and_track_delays` in the spec twice, and it should start the backoff over for the 2nd time rather than continuously increasing as the current implementation works.